File: C#/230. Kth Smallest Element in a BST.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    List<int> tree = new List<int>();
    public void convertInorder(TreeNode root){
        if(root==null) return;
        convertInorder(root.left);
        tree.Add(root.val);
        convertInorder(root.right);
    }
    public int KthSmallest(TreeNode root, int k) {
        convertInorder(root);
        return tree[k-1];
    }
}

File: C++/230. Kth Smallest Element in a BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> tree;
    void convertInorder(TreeNode* root){
        if(!root) return;
        convertInorder(root->left);
        tree.push_back(root->val);
        convertInorder(root->right);
    }
    int kthSmallest(TreeNode* root, int k) {
        convertInorder(root);
        return tree[k-1];
    }
};
